### PR TITLE
Persist profile information during user authentication with estonian ID

### PR DESCRIPTION
**Pull request by kstam**

### CODE REVIEW

File: src/infrastructure/idAuthentication/DokobitAuthenticationProvider.ts [hunk 1]
```diff
@@ -45,8 +49,29 @@ export class DokobitAuthenticationProvider implements AuthenticationProvider {
 
     return {
       code: sessionStatus.code,
-      firstName: sessionStatus.name,
-      lastName: sessionStatus.surname,
+      profile: mapToProfile(sessionStatus),
```
Comment (oliverviljamaa) [inline src/infrastructure/idAuthentication/DokobitAuthenticationProvider.ts:52]: Cool, wasn't even thinking about doing it on this level, but it's cleaner indeed.

File: src/infrastructure/idAuthentication/DokobitAuthenticationProvider.ts [hunk 2]
```diff
@@ -45,8 +49,29 @@ export class DokobitAuthenticationProvider implements AuthenticationProvider {
...
     };
   }
 }
+
+function mapToProfile(sessionStatus: DokobitSessionStatus): Profile {
+  var id = getAndValidateEstonianIdCode(sessionStatus);
+  id.validate();
```
Comment (oliverviljamaa) [inline src/infrastructure/idAuthentication/DokobitAuthenticationProvider.ts:59]: We're already validating in `getAndValidateEstonianIdCode`, is this necessary?

File: src/infrastructure/idAuthentication/DokobitAuthenticationProvider.ts [hunk 3]
```diff
@@ -45,8 +49,29 @@ export class DokobitAuthenticationProvider implements AuthenticationProvider {
...
+  }
+  return idCode;
+}
+
+function getSex(id: Isikukood): Sex {
+  const gender = id.getGender();
+  if (gender === 'unknown') {
+    throw new AuthenticationError('Non parsable gender');
+  }
+  return gender === 'male' ? Sex.MALE : Sex.FEMALE;
```
Comment (oliverviljamaa) [inline src/infrastructure/idAuthentication/DokobitAuthenticationProvider.ts:76]: Perhaps we could be more explicit with determining female, e.g.:
```suggestion
  const gender = id.getGender();
  
  switch (gender) {
    case 'male':
      return Sex.MALE;
    case 'female':
      return Sex.FEMALE;
    default:
      throw new AuthenticationError('Non parsable gender');    
  }
```

File: src/infrastructure/idAuthentication/DokobitAuthenticationProvider.ts [hunk 4]
```diff
@@ -6,7 +6,11 @@ import {
   AuthenticationSessionToken,
   AuthenticationResult,
 } from '../../domain/model/idAuthentication/models';
-import { DokobitClient } from './DokobitClient';
+import { DokobitClient, DokobitSessionStatus } from './DokobitClient';
+import { Profile } from '../../domain/model/user/Profile';
+import Isikukood from 'isikukood';
```
Comment (oliverviljamaa) [inline src/infrastructure/idAuthentication/DokobitAuthenticationProvider.ts:11]: Can we move this up with other external imports? 🙏

File: src/application/service/authentication/EstonianIdAuthenticator.test.ts [hunk 5]
```diff
@@ -11,14 +15,42 @@ describe(EstonianIdAuthenticator, () => {
...
       getExistingOrCreateNewUser,
-      generateAuthToken
+      generateAuthToken,
+      userRepository
     );
 
     await expect(authenticator.authenticate('some-dokobit-token')).rejects.toThrow(AuthenticationError);
   });
+
+  it('creates a user and fills the profile information', async () => {
```
Comment (oliverviljamaa) [inline src/application/service/authentication/EstonianIdAuthenticator.test.ts:28]: 👏

File: src/infrastructure/idAuthentication/DokobitAuthenticationProvider.ts (hunk 2 above)
Comment (kstam) [reply src/infrastructure/idAuthentication/DokobitAuthenticationProvider.ts:59]: oh.. missed this... good spot

File: src/infrastructure/idAuthentication/DokobitAuthenticationProvider.ts (hunk 3 above)
Comment (kstam) [reply src/infrastructure/idAuthentication/DokobitAuthenticationProvider.ts:76]: yeah, much cleaner

**Pull request merged**